### ISSUE TITLE
Generate code for query parameters

### ISSUE THREAD

**Issue by berkowitzi** [koxudaxi/datamodel-code-generator#1071]
First off, my apologies if this already supported in which case my ask would be better docs!

**Is your feature request related to a problem? Please describe.**
When I generate pydantic types from an openapi file that includes query parameters, no code is generated for parameter types. I'll use the example API in the usage docs as an example. In this case, nothing is generated for the listPets parameter(s).
```
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
  license:
    name: MIT
servers:
  - url: http://petstore.swagger.io/v1
paths:
  /pets:
    get:
      summary: List all pets
      operationId: listPets
      tags:
        - pets
      parameters:
        - name: limit
          in: query
          description: How many items to return at one time (max 100)
          required: false
          schema:
            type: integer
            format: int32
      responses:
        '200':
          description: A paged array of pets
          headers:
            x-next:
              description: A link to the next page of responses
              schema:
                type: string
          content:
            application/json:
              schema:
                $ref: "#/components/schemas/Pets"
```

**Describe the solution you'd like**
Code to be generated :)

**Describe alternatives you've considered**
Writing the types manually.

**Additional context**

**Comment by koxudaxi** [comment]
@berkowitzi 
Thank you for creating the issue.
This cli doesn't support the parameters.
We have discussed the topic  in https://github.com/koxudaxi/datamodel-code-generator/issues/433

What do you expect the result?
Is this model what you want? :thinking: 
```python
class PetsGetParameters(BaseModel):
  id: int
```

**Comment by hambrosia** [comment]
We were looking into this recently and stumbled onto this thread. We'd be interested in generating models for query parameters as well. The ability to go from something like below:

```
"/user/roles": {
            "get": {
                "description": "List roles available to currently logged in user.",
                "operationId": "ListRolesForUser",
                "parameters": [
                    {
                        "name": "nextToken",
                        "in": "query",
                        "description": "input token for pagination; when this value is provided as input, the service returns results from where the previous response left off.",
                        "schema": {
                            "type": "string",
                            "description": "input token for pagination; when this value is provided as input, the service returns results from where the previous response left off."
                        }
                    },
                    {
                        "name": "pageSize",
                        "in": "query",
                        "description": "maximum number of elements to include in the response",
                        "schema": {
                            "type": "integer",
                            "description": "maximum number of elements to include in the response",
                            "format": "int32"
                        }
                    },
                    {
                        "name": "omniFilterText",
                        "in": "query",
                        "description": "return any roles that match the input in accountFullName, accountShortName, accountAWSNumber, roleName",
                        "schema": {
                            "type": "string",
                            "description": "return any roles that match the input in accountFullName, accountShortName, accountAWSNumber, roleName"
                        }
                    },
                    {
                        "name": "accountAWSNumber",
                        "in": "query",
                        "description": "return any roles that are available for the specified AWS account number",
                        "schema": {
                            "type": "string",
                            "description": "return any roles that are available for the specified AWS account number"
                        }
                    }
                ],
```

And generate something similar to the following:

```
class RoleQuery(BaseModel):
    nextToken: Optional[int]
    pageSize: Optional[int]
    omniFilterText: Optional[str]
    accountAWSNumber: Optional[str]
```
    

Which could then be used in an endpoint to validate the query params. We could try to contribute it as well if it something maintainers are interested in having.

**Comment by koxudaxi** [comment]
@hambrosia

Thank you for your feedback.
If you have the time then please create the PR.

We can create models from  the return value of parse_item()
https://github.com/koxudaxi/datamodel-code-generator/blob/cdc2404748136e60db7899b19704e6a201415a7e/datamodel_code_generator/parser/openapi.py#L275-L285


Maybe we should add parameters to `OpenAPIScope`
https://github.com/koxudaxi/datamodel-code-generator/blob/f299b00c9d05c259eefa896d778e370e4c2d1373/datamodel_code_generator/__init__.py#L190-L193

Also, we can add your example to unittest.

**Comment by hambrosia** [comment]
Thanks @koxudaxi will be starting on this today.

**Comment by koxudaxi** [comment]
I have released a new version 0.17.1
The version has an option --openapi-scopes parameters
Thank you very much!!
